### DISCUSSION TITLE
Mixed JS & TS codebase errors on lint

### DISCUSSION THREAD

**Discussion by MylesKingsnorth** [eslint/eslint#16684] [Q&A]
Hey, I have a codebase that is mostly JS however as of recently we have decided to move to TS but we are replacing files as we go.

I have the following example component/folder structure:

```
apps:
    example:
        Components:
            Example.js
    shared:
        util.ts
```

Inside `Example.js` I'm importing a function from `util.ts`. My IDE reports no errors, when I compile my apps I get no errors, I have setup import aliases in my tsconfig etc, however when I run eslint in the terminal it errors 

> Cannot find module '/apps/shared/util' from '/apps/example/Components/Example'

Command `./node_modules/.bin/eslint --ext .js,.jsx,.ts,.tsx --format codeframe apps/example --quiet`

A stripped-back version of my `.eslintrc` looks like:

```json
{
    "globals": {...},
    "parser": "babel-eslint",
    "extends": [
        "react-app",
        "airbnb",
        "prettier",
        "plugin:react-hooks/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:import/typescript"
    ],
    "plugins": [
        "react",
        "prettier",
        "react-hooks",
        "eslint-plugin-import",
        "@typescript-eslint",
        "import"
    ],
    "rules": {
        "import/extensions": [
            "error",
            "ignorePackages",
            {
                "js": "never",
                "jsx": "never",
                "ts": "never",
                "tsx": "never"
            }
        ],
        ...
    },
    "settings": {
        "import/extensions": [".js", ".jsx", ".ts", ".tsx"],
        "import/parsers": {
            "@typescript-eslint/parser": [".ts", ".tsx", ".js", ".jsx"],
            "@typescript-eslint": [".ts", ".tsx", ".js", ".jsx"]
        },
        "import/resolver": {
            "jsconfig": {
                "config": "jsconfig.json"
            },
            "typescript": {
                "config": "tsconfig.json"
            },
            "node": {
                "extensions": [".js", ".jsx", ".ts", ".tsx"]
            }
        }
    },
    "parserOptions": {
        "project": ["./tsconfig.json"]
    }
}
```

I also tried to use overrides as per [this](https://github.com/typescript-eslint/typescript-eslint/issues/109#issuecomment-536160947)

```json
{
    "parser": "@babel/eslint-parser",
    "overrides": [
        {
            "files": ["**/*.ts", "**/*.tsx"],
            "env": { "browser": true, "es6": true, "node": true },
            "extends": [
                "plugin:@typescript-eslint/eslint-recommended",
                "plugin:@typescript-eslint/recommended",
                "plugin:import/typescript"
            ],
            "parser": "@typescript-eslint/parser",
            "plugins": ["react", "@typescript-eslint"],
            "parserOptions": {
                "ecmaFeatures": { "jsx": true },
                "ecmaVersion": 2018,
                "sourceType": "module",
                "project": "./tsconfig.json"
            }
        }
    ]
}
```

**Comment by mdjermanovic** [comment]
Hi @MylesKingsnorth! 

> Inside `Example.js` I'm importing a function from `util.ts`. My IDE reports no errors, when I compile my apps I get no errors, I have setup import aliases in my tsconfig etc, however when I run eslint in the terminal it errors
> 
> > Cannot find module '/apps/shared/util' from '/apps/example/Components/Example'

Is this a lint error reported by ESLint or a runtime error? Can you please copy & paste here the complete output you're getting?

**Reply by MylesKingsnorth** [reply]
Hi @mdjermanovic, apologies I should have verified before but it was a problem with a plugin I was using. Thanks anyways
